app/forms/questionnaire_form.py

```diff
@@ -134,18 +202,22 @@ def generate_form(schema, block_json, data, answer_store):
 
     class DynamicForm(QuestionnaireForm):
         date_ranges = []
+        calculated_question = []
```

Comment:
These 2 lines blow my mind!
We are setting these for every form, just because.

Reply:
What Jon meant to say was that since we have the block_json available to us at validate then do we need to create the date_ranges and calculated_question objects here?

Reply:
thanks adam

Reply:
We've got a separate card open now to get rid of `DynamicForm` anyway. Better to allow `QuestionnaireForm` to take these sort of values in its cctor and set its own bounds at that point.